Corona-Warn-App/src/main/java/de/rki/coronawarnapp/presencetracing/risk/PresenceTracingRiskRepository.kt

```diff
@@ -0,0 +1,118 @@
+package de.rki.coronawarnapp.presencetracing.risk
+
+import androidx.room.Dao
+import androidx.room.Entity
+import androidx.room.ForeignKey
+import androidx.room.Insert
+import androidx.room.PrimaryKey
+import androidx.room.Query
+import de.rki.coronawarnapp.eventregistration.storage.TraceLocationDatabase
+import de.rki.coronawarnapp.eventregistration.storage.entity.TraceLocationCheckInEntity
+import de.rki.coronawarnapp.risk.TraceLocationCheckInRisk
+import de.rki.coronawarnapp.util.TimeStamper
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import org.joda.time.Days
+import org.joda.time.Instant
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class PresenceTracingRiskRepository @Inject constructor(
+    private val presenceTracingRiskCalculator: PresenceTracingRiskCalculator,
+    private val databaseFactory: TraceLocationDatabase.Factory,
+    private val timeStamper: TimeStamper
+) {
+
+    private val database by lazy {
+        databaseFactory.create()
+    }
+
+    private val traceTimeIntervalMatchDao by lazy {
+        database.traceTimeIntervalMatchDao()
+    }
+
+    private val normalizedTime = traceTimeIntervalMatchDao.allEntries().map {
+        it.map {
+            it.toModel()
+        }
+    }.map {
+        presenceTracingRiskCalculator.calculateNormalizedTime(it)
+    }
+
+    val traceLocationCheckInRiskStates: Flow<List<TraceLocationCheckInRisk>> =
+        normalizedTime.map {
+            presenceTracingRiskCalculator.calculateCheckInRiskPerDay(it)
+        }
+
+    val presenceTracingDayRisk: Flow<List<PresenceTracingDayRisk>> =
+        normalizedTime.map {
+            presenceTracingRiskCalculator.calculateAggregatedRiskPerDay(it)
+        }
+
+    suspend fun replaceAllMatches(list: List<CheckInWarningOverlap>) {
+        deleteAllMatches()
+        addAll(list)
+    }
+
+    private suspend fun addAll(list: List<CheckInWarningOverlap>) {
+        traceTimeIntervalMatchDao.insert(list.map { it.toEntity() })
+    }
+
+    suspend fun deleteStaleMatches() {
+        val endTime = timeStamper.nowUTC.minus(Days.days(15).toStandardDuration())
+        traceTimeIntervalMatchDao.deleteOlderThan(endTime.millis)
+    }
+
+    suspend fun deleteAllMatches() {
+        traceTimeIntervalMatchDao.deleteAll()
+    }
+}
+
+@Dao
+interface TraceTimeIntervalMatchDao {
+
+    @Query("SELECT * FROM TraceTimeIntervalMatchEntity")
+    fun allEntries(): Flow<List<TraceTimeIntervalMatchEntity>>
+
+    @Query("DELETE FROM TraceTimeIntervalMatchEntity")
+    suspend fun deleteAll()
+
+    @Query("DELETE FROM TraceTimeIntervalMatchEntity WHERE endTimeMillis < :endTimeMillis")
+    suspend fun deleteOlderThan(endTimeMillis: Long)
+
+    @Insert
+    suspend fun insert(entities: List<TraceTimeIntervalMatchEntity>)
+}
+
+@Entity
```

Comment:
Use @ColumnInfo to specify the name

Reply:
What would I needed for? The attribute names are fine to be used as column names.

Reply:
If one just wants to adjust the name of a variable, you don't necessarily need to update the schema of the db or all handwritten sql queries. Same for r8 obfuscation